internal/app/api/handler/user/user.go

```diff
@@ -47,7 +52,12 @@ func (h *Handler) SignUp(w http.ResponseWriter, r *http.Request) {
 	err := json.NewDecoder(r.Body).Decode(&userSignup)
 
 	if err != nil {
-		respond.JSON(w, http.StatusInternalServerError, h.em.InvalidValue.Request)
+		respond.JSON(w, http.StatusBadRequest, h.em.InvalidValue.ValidationFailed)
```

Comment:
Log an error before response.
The Request error is okay, use can change the message if necessary